improver_tests/psychrometric_calculations/total_precipitable_water/test_total_precipitable_water.py

```diff
@@ -0,0 +1,211 @@
+    assert (
+        result.attributes["title"]
+        == "Global Enhanced Model Forecast on Global 10 km Standard Grid"
+    )
+    assert result.data.shape == mock_cube.data.shape
+
+
+def test_metadata_for_ukv_model(mock_cube):
+    """
+    Check that the metadata is preserved correctly for the UKV model input.
+    """
+    # Override title of mock_cube for UKV model test
+    mock_cube.attributes["title"] = (
+        "UKV Enhanced Model Forecast on UK 2 km Standard Grid"
+    )
+    plugin = PrecipitableWater()
+    result = plugin.process(mock_cube)
+    assert (
+        result.attributes["title"]
+        == "UKV Enhanced Model Forecast on UK 2 km Standard Grid"
+    )
+
+
+def test_pressure_conversion_and_shape(mock_cube):
+    """
+    Verify output shape matches input cube's horizontal dimensions.
+    """
+    plugin = PrecipitableWater()
+    result = plugin.process(mock_cube)
+    # Expect shape to match horizontal dimensions only
+    expected_shape = mock_cube.data.shape
+    assert result.data.shape == expected_shape
+
+
+def test_invalid_input_type():
+    """
+    Ensure plugin raises error when input is not a cube.
+    """
+    plugin = PrecipitableWater()
+    with pytest.raises(AttributeError):
+        plugin.process("not_a_cube")
+
+
+def test_output_units_and_standard_name(mock_cube):
+    """
+    Confirm output units and standard names are correctly set.
+    """
+    plugin = PrecipitableWater()
+    result = plugin.process(mock_cube)
+    assert result.units == "m"
+    assert result.standard_name == "lwe_thickness_of_precipitation_amount"
+
+
+def test_precipitable_water_known_values():
+    """
+    Test the plugin with a cube with known input humidity values and pressure levels to verify output.
+    """
+    pressure_levels = np.array([100000, 90000], dtype=np.float32)  # Pa
+    data = np.full((2, 2, 2), 0.005, dtype=np.float32)  # uniform humidity
+
+    cube = set_up_variable_cube(
+        data,
+        name="specific_humidity",
+        units="kg kg-1",
+        spatial_grid="latlon",
+        vertical_levels=pressure_levels,
+        pressure=True,
+        attributes={
+            "title": "Global Enhanced Model Forecast on Global 10 km Standard Grid"
+        },
+    )
+
+    plugin = PrecipitableWater()
+    result = plugin.process(cube)
+
+    # Expected calculation:
+    # delta_p = abs(100000 - 90000) = 10000 Pa
+    # TPW = 0.005 * 10000 / (9.80665 * 1000) ≈ 0.005096 m
+    expected_value = 0.005096
+    expected_array = np.full((2, 2, 2), expected_value, dtype=np.float32)
+
+    np.testing.assert_allclose(result.data, expected_array, rtol=1e-4, atol=1e-6)
+
+
+def test_process_with_pressure_coord(mock_cube):
+    """
+    Ensure plugin retrieves 'pressure' coordinate successfully when present.
+    """
+    plugin = PrecipitableWater()
+    result = plugin.process(mock_cube)
+    assert isinstance(result, iris.cube.Cube)
+    assert result.coord("pressure") is not None
```

Comment:
`Cube.coord("not_there")` will raise a `CoordinateNotFoundError`, so you can simplify this to:
```suggestion
    result.coord("pressure")
```